stdlib/locale.pyi

```diff
@@ -103,6 +103,15 @@ def atof(string: _str, func: Callable[[_str], float] = ...) -> float: ...
 def atoi(string: _str) -> int: ...
 def str(val: float) -> _str: ...
 
+# native gettext functions
+# https://github.com/python/cpython/blob/f4c03484da59049eb62a9bf7777b963e2267d187/Modules/_localemodule.c#L626
+if sys.platform == "linux" or sys.platform == "darwin":
+    def gettext(__msg: _str) -> _str: ...
+    def dgettext(__domain: _str, __msg: _str) -> _str: ...
+    def dcgettext(__domain: _str, __msg: _str, __category: _str) -> _str: ...
+    def textdomain(__domain: _str | None) -> _str | None: ...
+    def bindtextdomain(__domain: _str, __dir: _str) -> _str | None: ...
```

Comment:
The dir argument can also be a path-like object or None. You probably want to use `_typeshed.StrPath` but I haven't checked if bytes paths work.

Reply:
Added with `StrOrBytesPath | None`

Reply:
While it does take byte paths as input, if you use bytes that aren't a valid string it will successfully set the path to that then crash trying to return it because it always returns a str.

Reply:
In that case I think it's safer to just allow `StrPath` here.

Reply:
Changed to `StrPath`